### DISCUSSION TITLE
Custom Domain - Verification failed due to Certbot not being able to access .well-known

### DISCUSSION THREAD

**Discussion by yngwi** [appwrite/appwrite#1571] [Q&A] [answered]
Hi,

I'm trying out appwrite on my server using a slightly modified docker-compose file. I have replaced the included Traefik with my own instance already present on the server and replaced the HTTP endpoints with redirects to HTTPS. As far as I can tell everything works as expected (couldn't try websockets so far but the login demo app modified to use react hooks works fine, I can log in and out) but I'm not able to get a certificate for a custom domain. I have added the CNAME and it seems to verify just fine but Certbot fails with the following error message: 

```
Certbot failed to authenticate some domains (authenticator: webroot). The Certificate Authority reported these problems:
  Domain: [domain]
  Type:   unauthorized
  Detail: Invalid response from http://[domain]/.well-known/acme-challenge/NSc0sKKo6vKlA3AwiWCXui3x7UtLyXkF2RziUCpznEk [173.212.197.114]: 404
```

Is this something that is due to the custom Traefik or are there additional steps I have to take that the custom domain works? I'm grateful for any input, thank you in advance.

Best,
Daniel

**Comment by eldadfux** [comment, marked as answer]
Hey Daniel, Appwrite uses Traefik to server the custom SSL certificates (not generation just serving). You could potentially use the generated certificates together with your own proxy. The certificates are located at the appwrite-certificates volume.

**Reply by yngwi** [reply]
Hi, thank you for your reply! I'm not sure I follow you. Appwrite isn't able to generate certificates with LetsEncrypt because it has problems exposing .well-known on the custom domain. I assume this is because I try to use my own global Traefik instance. It creates the main certificate used for securing my main Appwrite domain which works fine so far, but I see now that this has some drawbacks that I didn't realize before. I wonder if there is a way to replicate what Appwrite does with its own Traefik instance with my global one or if it is possible to host the whole thing including the Appwrite Trafik behind my global Traefik. But still, I guess, certificate creation would be a problem. Are there any tips for doing this?

Just as a note: I am already exposing certificates from my global Traefik to other containers using a shared local folder to use for instance in a mail container setup. So I could maybe just manually add any new domains as host to the Appwrite traefik labels so my global instance also creates those certificates and just share my global certificate volume with Appwrite? 

You can see the differences in my setup on [diffchecker](https://www.diffchecker.com/S27tMNoz).

**Reply by eldadfux** [reply]
If you're using your own Traefik you can just ignore the Appwrite one and completely skip it. Let your global Traefik handle SSL for all domain like its doing by default, and add Appwrite routing rules for the `/v1/realtime` websocket.

**Reply by yngwi** [reply]
Thank you I'm going to try that 👍

**Reply by yngwi** [reply]
@eldadfux I'm sorry to ping you again but can you advise on how to avoid the `X-Fallback-Cookies` that seems to get used when the domain isn't verified. I have now set up my DNS and Traefik in a way that the whole api is just also reachable from the second domain so it wouldn't be a third party cookie any more and the fallback wouldn't have to be used but it only gets used when the project in Appwrite has a fully verified domain. My second domain actually is verified but still in progress as the "TLS" state doesn't leave "In Progress". Thank you.

**Reply by eldadfux** [reply]
@yngwi if you use the same top level domain for both your application and the Appwrite API the fallback cookie will not be used at all.

For example:
Your app: https://myapp.com
Your Appwrite API: https://api.myapp.com
